test/module/shared_model/builders/protobuf/transport_builder_test.cpp

```diff
@@ -0,0 +1,180 @@
+  uint64_t height;
+
+  std::function<void(const Error<std::string> &)> failCase =
+      [](const Error<std::string> &message) { FAIL() << message.error; };
+};
+
+/**
+ * @given valid proto object of transaction
+ * @when transport builder constructs model object from it
+ * @then original and built objects are equal
+ */
+TEST_F(TransportBuilderTest, TransactionCreationTest) {
+  auto orig_tx = createTransaction();
+  auto proto_tx = orig_tx.getTransport();
+
+  TransportBuilder<proto::Transaction, validation::DefaultTransactionValidator>
+      transport_builder(proto_tx);
+
+  auto built_tx = transport_builder.build();
+  built_tx.match(
+      [&orig_tx](const Value<Transaction> &tx) {
+        ASSERT_EQ(tx.value.getTransport().SerializeAsString(),
+                  orig_tx.getTransport().SerializeAsString());
+      },
+      failCase);
+}
+
+/**
+ * @given valid proto object of query
+ * @when transport builder constructs model object from it
+ * @then original and built objects are equal
+ */
+TEST_F(TransportBuilderTest, QueryCreationTest) {
+  auto orig_query = createQuery();
+  auto proto_query = orig_query.getTransport();
+
+  TransportBuilder<proto::Query, validation::DefaultQueryValidator>
+      transport_builder(proto_query);
+
+  auto built_query = transport_builder.build();
+  built_query.match(
+      [&orig_query](const Value<Query> &query) {
+        ASSERT_EQ(query.value.getTransport().SerializeAsString(),
+                  orig_query.getTransport().SerializeAsString());
+      },
+      failCase);
+}
+
+/**
+ * @given valid proto object of block
+ * @when transport builder constructs model object from it
+ * @then original and built objects are equal
+ */
+TEST_F(TransportBuilderTest, BlockCreationTest) {
+  auto orig_block = createBlock();
+  auto proto_block = orig_block.getTransport();
+
+  TransportBuilder<proto::Block, validation::DefaultBlockValidator>
+      transport_builder(proto_block);
+
+  auto built_block = transport_builder.build();
+  built_block.match(
+      [&orig_block](const Value<Block> &block) {
+        ASSERT_EQ(block.value.getTransport().SerializeAsString(),
+                  orig_block.getTransport().SerializeAsString());
+      },
+      failCase);
+}
+
+/**
+ * @given valid proto object of proposal
+ * @when transport builder constructs model object from it
+ * @then original and built objects are equal
+ */
+TEST_F(TransportBuilderTest, ProposalCreationTest) {
+  auto orig_proposal = createProposal();
+  auto proto_proposal = orig_proposal.getTransport();
+
+  TransportBuilder<proto::Proposal, validation::DefaultProposalValidator>
+      transport_builder(proto_proposal);
+
+  auto built_proposal = transport_builder.build();
+  built_proposal.match(
+      [&orig_proposal](const Value<Proposal> &proposal) {
+        ASSERT_EQ(proposal.value.getTransport().SerializeAsString(),
+                  orig_proposal.getTransport().SerializeAsString());
+      },
+      failCase);
+}
```

Comment:
Need to add tests where objects will be not created.

Reply:
Yes, creation from invalid transport objects might be included